packages/@o3r/schematics/src/rule-factories/check-packages-peers/index.ts

```diff
@@ -1,15 +1,124 @@
+import type { LoggerApi } from '@angular-devkit/core/src/logger';
+
+/** Interface containing a npm package name, needed version and optionally found version */
+interface PackageVersion {
+  /** Npm package name */
+  packageName: string;
+  /** Npm package needed version */
+  version: string;
+  /** Npm package installed version found */
+  foundVersion?: string;
+}
+
+/**
+ * Get package manager used in runs
+ * Defaults to the package manager setup in process.env if no package manager set in angular.json
+ *
+ * @param angularJsonString Content of angular.json file
+ */
+function getPackageManager(angularJsonString?: string | null) {
+  let packageManager = process.env && process.env.npm_execpath && process.env.npm_execpath.indexOf('yarn') === -1 ? 'npm' : 'yarn';
+  if (angularJsonString) {
+    const angularJsonObj = commentJson.parse(angularJsonString) as any;
+    if (angularJsonObj?.cli?.packageManager) {
+      packageManager = angularJsonObj.cli.packageManager;
+    }
+  }
+  return packageManager;
+}
+
+
+/**
+ * Check if the first level of peer deps of a given package are installed.
+ * List all not installed packages or packages with version mismatch
+ *
+ * @param packageName
+ */
+function getPackagesToInstallOrUpdate(packageName: string) {
+  let installedPackage: PackageJson;
+  try {
+    installedPackage = require(`${packageName}${path.posix.sep}package.json`);
+  } catch (err) {
+    throw new Error(`The provided package is not installed: ${packageName}`);
+  }
+
+  const packagesToInstall: PackageVersion[] = [];
+  const packagesWrongVersion: PackageVersion[] = [];
+
+  const optionalPackages = Object.entries(installedPackage.peerDependenciesMeta || {})
+    .filter(([, dep]) => dep?.optional)
+    .map(([depName]) => depName);
+  const peerDependenciesMap = Object.entries(installedPackage.peerDependencies || {})
+    .reduce<Partial<Record<string, string>>>((acc, [name, val]) => {
+      if (!optionalPackages.includes(name)) {
+        acc[name] = val;
+      }
+      return acc;
+    }, {});
+  Object.entries(peerDependenciesMap).forEach(([pName, pVersion]) => {
+    let installedPackageVersion: string | undefined;
+    try {
+      installedPackageVersion = require(`${pName}${path.posix.sep}package.json`).version;
+    } catch (err) {
+      packagesToInstall.push({ packageName: pName, version: pVersion! });
+    }
+    if (installedPackageVersion && !satisfies(installedPackageVersion, pVersion!)) {
+      packagesWrongVersion.push({ packageName: pName, foundVersion: installedPackageVersion, version: pVersion! });
+    }
+  });
+  return { packagesToInstall, packagesWrongVersion };
+}
+
+/**
+ * Log an instruction with the packages to install or update to match a package peer dependencies
+ *
+ * @param packageName
+ * @param angularJsonString
+ * @param logger
+ */
+export function checkPackagesToInstallOrUpdate(packageName: string, logger: LoggerApi, angularJsonString?: string | null) {
```

Comment:
if not used outside this file, is it worth exporting it?

Reply:
I hesitated by I finally kept the export because it was exported before and because it was not a big deal to do so

Reply:
Well, this way, next time we want to remove it, we will have to deprecate it because we won't know if anyone else is using it. Let's keep it as a private method and, if we ever need it outside, we can export it?